### DISCUSSION TITLE
Status conditions such as ClusterReady as status attributes?

### DISCUSSION THREAD

**Discussion by kriswuollett** [rabbitmq/cluster-operator#998] [Q&A] [answered]
I'm looking to create the CRD via Terraform and use a [`wait_for`](https://registry.terraform.io/providers/hashicorp/kubernetes/latest/docs/resources/manifest#using-wait_for-to-block-create-and-update-calls) condition with a [`depends_on`](https://www.terraform.io/language/meta-arguments/depends_on), e.g.,

```hcl
wait_for = {
  fields = {
    "status.phase" = "Running"
    "status.containerStatuses[0].ready" = "true",
  }
}
```

 I don't think it is possible to write a condition to search an array like was done for [printing the values](https://github.com/rabbitmq/cluster-operator/blob/b46b298874ed5bf19355a86317737378d6253fac/config/crd/bases/rabbitmq.com_rabbitmqclusters.yaml#L31).

So is an enhancement needed to be able to specifically look for a "status" such as `ClusterReady` or `AllReplicasReady`?

**Comment by Zerpet** [comment, marked as answer]
Not a great solution, but you could assume that `status.conditions[0]` is always `AllReplicasReady` type, which is always true given how we set conditions: https://github.com/rabbitmq/cluster-operator/blob/b46b298874ed5bf19355a86317737378d6253fac/api/v1beta1/rabbitmqcluster_status.go#L84-L89

We have no plans to change the status conditions in the future.
